wdl/Utils.wdl

```diff
@@ -758,3 +758,84 @@ task TransferVcfAnnotations {
     File annotated_vcf_index = output_file_name + ".tbi"
   }
 }
+
+
+task RenameVcfSamples {
+  input {
+    File vcf
+    File vcf_idx
+    File sample_renaming_json  # JSON with map from old sample IDs to new sample IDs
+    String samtools_cloud_docker
+    String renamed_suffix = "renamed_samples"
+    RuntimeAttr? runtime_attr_override
+  }
+
+  parameter_meta {
+      vcf: {
+        localization_optional: true
+      }
+  }
+
+  String vcf_renamed_samples_filename = basename(vcf, ".vcf.gz") + "_~{renamed_suffix}.vcf.gz"
+  String vcf_renamed_samples_index_filename = vcf_renamed_samples_filename + ".tbi"
+
+  # Disk must be scaled proportionally to the size of the VCF
+  Float input_size = size(vcf, "GiB")
+  RuntimeAttr default_attr = object {
+    mem_gb: 2.0,
+    disk_gb: ceil(10.0 + input_size),
+    cpu_cores: 1,
+    preemptible_tries: 3,
+    max_retries: 1,
+    boot_disk_gb: 10
+  }
+  RuntimeAttr runtime_attr = select_first([runtime_attr_override, default_attr])
+
+  command <<<
+    set -euo pipefail
+
+    # symlink vcf_index to current working dir
+    ln -s ~{vcf_idx} .
+
+    echo "Get original VCF samples"
+    export GCS_OAUTH_TOKEN=`gcloud auth application-default print-access-token`
+    bcftools query -l ~{vcf} > original_vcf_samples.list
+
+    echo "rename samples according to JSON map, getting the new corresponding list of sample IDs"
+    python - <<____EoF
+import json
+with open("~{sample_renaming_json}", 'r') as f_in:
+  sample_ids_map = json.load(f_in)
+
+with open("original_vcf_samples.list", 'r') as f_in, open("renamed_samples.list", 'w') as f_out:
+  for line in f_in:
+    sample_id = line.rstrip()
+    renamed_id = sample_ids_map.get(sample_id, sample_id)
+    f_out.write(renamed_id + "\n")
+____EoF
+
+    echo "use bcftools reheader to change the sample names"
+    bcftools reheader \
+      -s "renamed_samples.list" \
```

Comment:
It seems like it would be simpler to just take this TSV sample ID map as the input instead - what do you think? Personally I prefer the TSV because I think it's easier to generate for an analyst, and it removes some computation from this task

Reply:
bcftools is taking as input just a list of the new sample IDs, which it will use based on position alone (no mapping involved). The issue I faced in designing this is that we have potentially multiple input VCFs, in which case I could either have multiple input sample-renaming files, or one master input file that we could subset. I decided the first approach was actually harder to deal with, so I went with the later. Both have their downsides though. That said, I think the computation is only adding mental overhead for understanding this task, I don't think it adds meaningfully to runtime.

I actually developed a script to produce this input file, which I considered adding to the PR, but I didn't call it in a WDL. I could add it to the `scripts` folder though.

Reply:
bcftools also has the option to take a whitespace-separated file with a column for the old ID & a column for the new ID, so that's what I was referring to - sorry for the confusion since you were using a different format here. If we stick with the VCF reheader approach, my personal preference would be to provide a TSV in that format as input, rather than a JSON, because it's super easy to generate for an analyst working in excel or with Terra data tables. 

My understanding is that the multiple input VCFs should contain the same samples, but they may be sharded by genomic position, so I think it's ok to have one reheader file for all VCFs.

Reply:
Used a totally different strategy (as discussed), but the input is now a TSV. The script that produces it is `gatk-sv/scripts/inputs/get_rename_benchmark_samples_map.py`